### DISCUSSION TITLE
Local render doesn't have primer-spec formatting

### DISCUSSION THREAD

**Discussion by lslavice** [eecs485staff/primer-spec#203] [Q&A] [answered]
Hello there! ENGR 101 is attempting to join the "make our project specs look pretty too" family, and I'm trying to do a local preview of a project spec. I'm trying to follow the instructions to do so, but I just seem to get basic markdown --> html rendering in the `_site` version of the specs. No sidebar navigation or pretty formatting.

Can you please pretend I'm like 10 years old and don't know what I'm doing, because that is not far off hahaha. 

If I am in the `docs` folder, and I run this command:
`bundle exec jekyll serve`

Then I get this warning: 
`Build Warning: Layout 'spec' requested in projects/matlab-practice-project/specs.md does not exist.`

Which I think is the problem, but I am clearly missing some step that maybe should be obvious. An html version of the spec's `.md` file is generated, but it just looks like this: 

<img width="470" alt="image" src="https://user-images.githubusercontent.com/70032904/186655383-6e06ed6d-e645-458b-8315-192184f5e0d3.png">

Any advice would be much appreciated!

**Comment by awdeorio** [comment]
Just a guess here: did you configure `docs/_config.yml`?

**Reply by lslavice** [reply]
aha.... I had the `_config.yml` file, but I had it one level up from `docs`. I misunderstood what was meant by "the root directory" lol. Thanks! :)

**Comment by seshrs** [comment, marked as answer]
Hi @lslavice, thanks for considering using Primer Spec for ENGR 101! Feel free to ask here / email me any time you need support :)

Like @awdeorio mentioned, I suspect the issue is that you placed your `_config.yml` file in a different directory than where you ran `bundle exec jekyll serve`. If you move the file to your `docs/` directory and try running the command again, Jekyll should pick up the config and render your site correctly. 

(I’ll need to update the instructions in step 3 of https://github.com/eecs485staff/primer-spec#usage to more clearly define what “site’s root directory” means. Thanks for the feedback!)

---

Tangent: If you’re interested in integrating [Primer Spec Preview](https://github.com/seshrs/upload-to-primer-spec-preview) to make it easier to preview changes to your specs on your GitHub PRs (without having to build locally), I can help you get that set up via email.

**Reply by lslavice** [reply]
Yes, that's what it was: I just misunderstood what was meant by the "site's root directory". Probably makes sense to someone who does this more often than I do! :) 

So, I can get everything to look PRETTYYYY!!!! when it's live on GitHub, but I can't get the local preview to work. I looked into some of the issues, and as far as I can tell it's because dealing with Ruby on Mac is a mess and a half. So, yes, if you have a way to do local previews without me spending hours fighting Ruby, I would appreciate it! 

Super excited to move our project specs off of google docs and to something a lot more navigable!!

**Reply by seshrs** [reply]
@lslavice I’ve sent you an email to get started integrating Primer Spec Preview with your GitHub repos!

Like @awdeorio mentioned below, it would be nice to figure out how to fix your setup, so any error messages / behaviors you notice would be helpful for us to debug your setup and improve the Primer Spec instructions :)

**Comment by awdeorio** [comment]
I'm on Mobile but I'll respond as best I can. I want to make it easy for
others to use this! Try using Homebrew to install. Ruby. Then restart your
terminal.

**Comment by lslavice** [comment]
hahaha, I did use homebrew to (re)install Ruby, and now things seem to be
even worse. Anyways! Sesh and I have an email chain going, and maybe once
the start of semester settles down, I can sit down with one of you in
person and you can see what kinds of errors I get. I have enough experience
with latex and markdown to be able to "visualize" the final rendering in my
head pretty well, so doing a few pull requests as I go to verify isn't a
huge deal for now. But yes! I'd love to help improve the user experience in
any way I can! Just might have to wait a few weeks hahaha


----
Laura K. Alford, PhD (she/her/hers)
Lecturer and Research Investigator
Naval Architecture & Marine Engineering
University of Michigan
